static/js/services/http-wrapper.js

```diff
@@ -32,13 +37,13 @@ var _ = require('underscore');
     '$http', '$q', 'ActiveRequests',
     function($http, $q, ActiveRequests) {
 
-      var wrap = function(args, fn) {
+      var wrap = function(args, fn, updateTarget) {
```

Comment:
Would you consider using the state name ('messages.details', etc), rather than taking in the `updateTarget` ('filters', 'left', etc)? Then on `cancelAll` you can check if the new state is a child of the target state.